modules/core/02-client/types/msgs.go

```diff
@@ -265,6 +269,32 @@ func (msg *MsgRecoverClient) ValidateBasic() error {
 	return nil
 }
 
+// NewMsgProvideCounterparty creates a new MsgProvideCounterparty instance
+func NewMsgProvideCounterparty(signer, clientID, counterpartyID string) *MsgProvideCounterparty {
```

Comment:
We need to also be able to provide the MerklePathPrefix here. Seems I didn't do that in my initial work.